### PR TITLE
HW7

### PR DESCRIPTION
**Pull request by fabl3ss**

### CODE REVIEW

File: crypto-service/src/loggers/zap_rabbitmq_logger.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		Port:     os.Getenv(config.EnvRabbitMQPort),
+		Username: os.Getenv(config.EnvRabbitMQUser),
+		Password: os.Getenv(config.EnvRabbitMQPassword),
+		Exchange: os.Getenv(config.EnvRabbitMQExchange),
+	}
+}
+
+func dialWithRabbitMQ(cfg *rabbitMQConfig) *amqp.Connection {
+	rabbitURL := fmt.Sprintf(
+		"amqp://%s:%s@%s:%s/",
```
Comment (NikitaMasych) [inline crypto-service/src/loggers/zap_rabbitmq_logger.go:110]: Maybe it will be better to make amqp url format constant, defined somewhere in this file?

**Review by NikitaMasych — APPROVED** [review]
# Abstract:
Logger✅
Message-broker✅
Consumer✅

# Docker:
I see that you remove auxiliary files using CLI in order to optimise image size. 
This, however, is not the main memory-overwhelming part.
Much more place take previous layers, what are composed one over one, thereby carrying redundant data. 
You may modify your instructions set for multi-staged build and use lighter base images. 
I made several tests on your project, results for crypto-service:

1.72 GB   - golang:latest, one staged (current version)
1.08 GB   - golang:alpine, one staged
1.02 GB   - golang:latest, build+prd staged
379.26 MB - golang:alpine, build+prd staged 
29.4 MB   - gcr.io/distroless/static-debian11, build+prd staged - this one is using special google image, very light and secure

# Architecture:
I think that logger had better to be in the application layer and not in domain.

# Logs amount:
You have nice logging system, however, you almost don't use it. 
I certainly think that you should increase logs amount.

File: crypto-service/src/loggers/zap_rabbitmq_logger.go (hunk 1 above)
Comment (fabl3ss) [reply crypto-service/src/loggers/zap_rabbitmq_logger.go:110]: Specifically in this example, it seems to me that using a template inside sprintf makes the code easier to understand.
Thanks for the tip, that's a good idea, but I'm not sure what exactly in this case.

**Comment by Bogdaan** [comment]
Summary:
1. ✅ Found docker-compose, broker, producer, consumer
2. Nice review, multi-stage builds is much more better 👍

**Pull request merged**